packages/syft/src/syft/store/sqlite_document_store.py

```diff
@@ -100,102 +66,53 @@ def __init__(
-
-    @property
-    def cur(self) -> sqlite3.Cursor:
-        if cache_key(self.db_filename) not in SQLITE_CONNECTION_POOL_CUR:
-            SQLITE_CONNECTION_POOL_CUR[cache_key(self.db_filename)] = self.db.cursor()
-
-        return SQLITE_CONNECTION_POOL_CUR[cache_key(self.db_filename)]
-
-    def _close(self) -> None:
-        self._commit()
-        REF_COUNTS[cache_key(self.db_filename)] -= 1
-        if REF_COUNTS[cache_key(self.db_filename)] <= 0:
-            # once you close it seems like other object references can't re-use the
-            # same connection
-            self.db.close()
-            del SQLITE_CONNECTION_POOL_DB[cache_key(self.db_filename)]
-        else:
-            # don't close yet because another SQLiteBackingStore is probably still open
+        with self._execute(
+            f"create table if not exists {self.table_name} (uid VARCHAR(32) NOT NULL PRIMARY KEY, "  # nosec
+            + "repr TEXT NOT NULL, value BLOB NOT NULL, "  # nosec
+            + "sqltime TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL)"
+        ) as _:
             pass
 
-    def _commit(self) -> None:
-        self.db.commit()
-
+    @contextmanager
     def _execute(
         self, sql: str, *args: list[Any] | None
-    ) -> Result[Ok[sqlite3.Cursor], Err[str]]:
-        with self.lock:
-            cursor: sqlite3.Cursor | None = None
-            # err = None
-            try:
-                cursor = self.cur.execute(sql, *args)
-            except Exception as e:
-                raise_exception(self.table_name, e)
-
-            # TODO: Which exception is safe to rollback on?
-            # we should map out some more clear exceptions that can be returned
-            # rather than halting the program like disk I/O error etc
-            # self.db.rollback()  # Roll back all changes if an exception occurs.
-            # err = Err(str(e))
-            self.db.commit()  # Commit if everything went ok
-
-            # if err is not None:
-            #     return err
-
-            return Ok(cursor)
+    ) -> Generator[sqlite3.Cursor, None, None]:
+        timeout = (
+            self.store_config.client_config.timeout
+            if self.store_config.client_config
+            else 5
+        )
+        con = sqlite3.connect(
+            self.file_path,
+            timeout=timeout,
+            check_same_thread=False,  # do we need this if we use the lock?
+            # check_same_thread=self.store_config.client_config.check_same_thread,
+        )
+        # Set journal mode to WAL.
+        con.execute("PRAGMA journal_mode = WAL")
+        con.execute("PRAGMA busy_timeout = 5000")
+        con.execute("PRAGMA temp_store = 2")
+        con.execute("PRAGMA synchronous = 1")
+
+        cur = con.cursor()
+        try:
+            yield cur.execute(sql, *args)
+            con.commit()
+        except Exception as e:
+            con.close()
+            raise e
+        finally:
+            con.close()
```

Comment:
Suggestion: would be a bit nicer and safer this way since the `con.execute("PRAGMA journal_mode = WAL")... cur = con.cursor()` part would be covered by the `con` context manager as well.

```suggestion
        with contextlib.closing(
            sqlite3.connect(
                self.file_path,
                timeout=timeout,
                check_same_thread=False,  # do we need this if we use the lock?
                # check_same_thread=self.store_config.client_config.check_same_thread,
            )
        ) as con:
            # Set journal mode to WAL.
            con.execute("PRAGMA journal_mode = WAL")
            con.execute("PRAGMA busy_timeout = 5000")
            con.execute("PRAGMA temp_store = 2")
            con.execute("PRAGMA synchronous = 1")

            cur = con.cursor()

            yield cur.execute(sql, *args)
            try:
                con.commit()
            finally:
                cur.close()
```

Reply:
Also at this point should we just `yield con.cursor()` and write
```py
with self._execute() as cur:
    cur.execute(insert_sql, [str(key), _repr_debug_(value), data, str(key)])
```
(and maybe change `def _execute` to `def _cursor`)
instead of
```py
with self._execute(...) as _:
    pass
```
?